Cs falcon enhancement (#26136)
demisto/content: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/README.md

- Alerts - Read and Write
- Hosts - Read
- IOC Manager - Read and Write

> - On-Demand Scans (ODS) - Read and Write

Reviewer: need to generate readme here. We should add a command to the command examples file (examples.txt) to do it. lets discuss

Author: will do

> - On-Demand Scans (ODS) - Read and Write

Reviewer: add to Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon_description.md

Author: done

Newly fetched incidents or detections will be mirrored in the chosen direction.  However, this selection does not affect existing incidents.

**Important Notes**

> - To ensure the mirroring works as expected, mappers are required, both for incoming and outgoing, to map the expected fields in XSOAR and CrowdStrike Falcon.

Reviewer 2: ```suggestion
- To ensure the mirroring works as expected, mappers are required, both for incoming and outgoing, to map the expected fields in Cortex XSOAR and CrowdStrike Falcon.
```

> - When *mirroring in* incidents from CrowdStrike Falcon to XSOAR:

Reviewer 2: ```suggestion
- When *mirroring in* incidents from CrowdStrike Falcon to Cortex XSOAR:
```

#### Human Readable Output

>### CrowdStrike Falcon file le10098bf0e311e989190662caec3daa_a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1

>|CreatedBy|CreatedTime|Description|ID|ModifiedBy|ModifiedTime|Name|Permission|SHA256|Type|
>|---|---|---|---|---|---|---|---|---|---|
>| spongobob@demisto.com | 2019-10-17T13:41:48.487520845Z | Demisto | le10098bf0e311e989190662caec3daa_a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | spongobob@demisto.com | 2019-10-17T13:41:48.487521161Z | Demisto | private | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | script |

### 12. cs-falcon-list-files

---

> Returns Returns a list of put-file ID's that are available for the user in the `put` command.

Reviewer 2: ```suggestion
Returns a list of put-file ID's that are available for the user in the `put` command.
```

                    26405458936820,
                    26405458936825,
                    26405458936826,
                    26405458936827,
                    26405458936832,
                    26405458936833,
                    26405458936834,
                    26405458936840,
                    26405458936841,
                    26405458936842,
                    26456998543793,
                    26456998544045,
                    26456998543652,
                    26456998543653,
                    26456998543656,
                    26456998543654,
                    26456998543950,
                    26456998543963
                ],
                "preemption_priority": 1,
                "profile_id": "a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1",
                "quarantine": true,
                "scan_completed_on": "2023-03-12T14:57:37.338506965Z",
                "scan_inclusions": [
                    "*"
                ],
                "scan_started_on": "2023-03-12T14:54:45.210172175Z",
                "sensor_ml_level_detection": 3,
                "sensor_ml_level_prevention": 3,
                "status": "failed"
            },
            {

>                 "cid": "a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1",

Author: will do

#### Context Output

There is no context output for this command.

#### Command example

```!cs-falcon-apply-quarantine-file-action filename=nc.exe action=delete comment=demo-comment```

#### Human Readable Output

>The Quarantined File with IDs ['a123456_sha123456'] was successfully updated.

### cs-falcon-ods-query-scan

***
Retrieve ODS scan details.

#### Base Command

`cs-falcon-ods-query-scan`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| filter | Valid CS-Falcon-FQL filter to query with. | Optional | 

> | ids | Comma-separated list of scan IDs to retrieve details about. If set will override all other arguments. | Optional | 

Reviewer 2: ```suggestion
| ids | Comma-separated list of scan IDs to retrieve details about. If set, will override all other arguments. | Optional | 
```

| initiated_from | Comma-separated list of scan initiation sources to filter by. | Optional | 
| status | Comma-separated list of scan statuses to filter by. | Optional | 
| severity | Comma-separated list of scan severities to filter by. | Optional | 

> | scan_started_on | UTC-format time of scan start to filter by. | Optional | 

Reviewer 2: ```suggestion
| scan_started_on | UTC-format start time of the scan to filter by. | Optional | 
```

> | scan_completed_on | UTC-format time of scan start to filter by. | Optional | 

Reviewer 2: ```suggestion
| scan_completed_on | UTC-format completion time of the scan to filter by. | Optional | 
```

> | offset | Starting index of overall result set from which to return ids. | Optional | 

Reviewer 2: ```suggestion
| offset | Starting index of overall result set from which to return IDs. | Optional | 
```

| limit | Maximum number of resources to return. | Optional | 

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |

> | CrowdStrike.ODSScan.id | String | a unique identifier for the scan event. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.id | String | A unique identifier for the scan event. | 
```

> | CrowdStrike.ODSScan.cid | String | a unique identifier for the client that triggered the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.cid | String | A unique identifier for the client that triggered the scan. | 
```

> | CrowdStrike.ODSScan.profile_id | String | a unique identifier for the scan profile used in the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.profile_id | String | A unique identifier for the scan profile used in the scan. | 
```

> | CrowdStrike.ODSScan.description | String | the ID of the description of the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.description | String | The ID of the description of the scan. | 
```

| CrowdStrike.ODSScan.scan_inclusions | String | The files or folders included in the scan. | 
| CrowdStrike.ODSScan.initiated_from | String | The source of the scan initiation. | 

> | CrowdStrike.ODSScan.quarantine | Boolean | If was set to quarantine. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.quarantine | Boolean | Whether the scan was set to quarantine. | 
```

| CrowdStrike.ODSScan.cpu_priority | Number | The CPU priority for the scan \(1-5\). | 
| CrowdStrike.ODSScan.preemption_priority | Number | The preemption priority for the scan. | 
| CrowdStrike.ODSScan.metadata.host_id | String | A unique identifier for the host that was scanned. | 
| CrowdStrike.ODSScan.metadata.host_scan_id | String | A unique identifier for the scan that was performed on the host. | 
| CrowdStrike.ODSScan.metadata.scan_host_metadata_id | String | A unique identifier for the metadata associated with the host scan. | 
| CrowdStrike.ODSScan.metadata.filecount.scanned | Number | The number of files that were scanned. | 
| CrowdStrike.ODSScan.metadata.filecount.malicious | Number | The number of files that were identified as malicious. | 
| CrowdStrike.ODSScan.metadata.filecount.quarantined | Number | The number of files that were quarantined. | 
| CrowdStrike.ODSScan.metadata.filecount.skipped | Number | The number of files that were skipped during the scan. | 
| CrowdStrike.ODSScan.metadata.filecount.traversed | Number | The number of files that were traversed during the scan. | 

> | CrowdStrike.ODSScan.metadata.status | String | The status of the scan on this host. \(e.q., "pending", "running", "completed", or "failed"\). | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.metadata.status | String | The status of the scan on this host \(e.g., "pending", "running", "completed", or "failed"\). | 
```

| CrowdStrike.ODSScan.metadata.started_on | Date | The date and time that the scan started. | 
| CrowdStrike.ODSScan.metadata.completed_on | Date | The date and time that the scan completed. | 
| CrowdStrike.ODSScan.metadata.last_updated | Date | The date and time that the metadata was last updated. | 

> | CrowdStrike.ODSScan.status | String | The status of the scan. \(e.q., "pending", "running", "completed", or "failed"\). | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.status | String | The status of the scan \(e.g., "pending", "running", "completed", or "failed"\). | 
```

| CrowdStrike.ODSScan.hosts | String | A list of the host IDs that were scanned. | 
| CrowdStrike.ODSScan.endpoint_notification | Boolean | A boolean value indicating whether endpoint notifications are enabled. | 

> | CrowdStrike.ODSScan.pause_duration | Number | The number of minutes to pause between scanning each file in hours. | 

Reviewer 2: Not sure about this description.
Its the "number of minutes", but what does it mean "in hours"?

| CrowdStrike.ODSScan.max_duration | Number | The maximum amount of time to allow for the scan job in hours. | 
| CrowdStrike.ODSScan.max_file_size | Number | The maximum file size \(in MB\) to scan. | 
| CrowdStrike.ODSScan.sensor_ml_level_detection | Number | The level of detection sensitivity for the local sensor machine learning model. | 
| CrowdStrike.ODSScan.sensor_ml_level_prevention | Number | The level of prevention sensitivity for the local sensor machine learning model. | 
| CrowdStrike.ODSScan.cloud_ml_level_detection | Number | The level of detection sensitivity for the cloud machine learning model. | 
| CrowdStrike.ODSScan.cloud_ml_level_prevention | Number | The level of prevention sensitivity for the cloud machine learning model. | 

> | CrowdStrike.ODSScan.policy_setting | Number | a list of policy setting IDs for the scan job \(these correspond to specific policy settings in the Falcon console\). | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.policy_setting | Number | A list of policy setting IDs for the scan job \(these correspond to specific policy settings in the Falcon console\). | 
```

                "sensor_ml_level_detection": 3,
                "sensor_ml_level_prevention": 3,
                "status": "failed"
            }
        ]
    }
}
```

#### Human Readable Output

>### CrowdStrike Falcon ODS Scans

>|ID|Status|Severity|File Count|Description|Hosts/Host groups|End time|Start time|Run by|
>|---|---|---|---|---|---|---|---|---|
>| a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | failed |  | scanned: 0<br/>malicious: 0<br/>quarantined: 0<br/>skipped: 0<br/>traversed: 518464 | desc3456346 | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 |  | 2023-03-15T15:57:37.59543591Z | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 |
>| a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | failed |  | scanned: 0<br/>malicious: 0<br/>quarantined: 0<br/>skipped: 0<br/>traversed: 518511 |  | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | 2023-03-13T14:50:26.259846586Z | 2023-03-13T14:47:44.674818897Z | someone@email.com |

### cs-falcon-ods-query-scheduled-scan

***
Retrieve ODS scheduled scan details.

#### Base Command

`cs-falcon-ods-query-scheduled-scan`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| filter | Valid CS-Falcon-FQL filter to query with. | Optional | 

> | ids | Comma-separated list of scan IDs to retrieve details about. If set will override all other arguments. | Optional | 

Reviewer 2: ```suggestion
| ids | Comma-separated list of scan IDs to retrieve details about. If set. will override all other arguments. | Optional | 
```

| initiated_from | Comma-separated list of scan initiation sources to filter by. | Optional | 
| status | Comma-separated list of scan statuses to filter by. | Optional | 
| created_on | UTC-format time of scan creation to filter by. | Optional | 
| created_by | UTC-format time of scan creator to filter by. | Optional | 

> | start_timestamp | UTC-format time of scan start to filter by. | Optional | 

Reviewer 2: ```suggestion
| start_timestamp | UTC-format start time of scan to filter by. | Optional | 
```

| deleted | Deleted scans only. | Optional | 

> | offset | Starting index of overall result set from which to return ids. | Optional | 

Reviewer 2: ```suggestion
| offset | Starting index of overall result set from which to return IDs. | Optional | 
```

                "start_timestamp": "2023-05-20T06:49"
            },
            "sensor_ml_level_detection": 2,
            "sensor_ml_level_prevention": 2,
            "status": "scheduled"
        }
    }
}
```

#### Human Readable Output

>### CrowdStrike Falcon ODS Scheduled Scans

>|ID|Hosts targeted|Description|Host groups|Start time|Created by|
>|---|---|---|---|---|---|
>| a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | 1 |  | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | 2023-05-20T06:49 | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 |

### cs-falcon-ods-query-scan-host

***
Retrieve ODS scan host details.

#### Base Command

`cs-falcon-ods-query-scan-host`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| filter | Valid CS-Falcon-FQL filter to query with. | Optional | 
| host_ids | Comma-separated list of host IDs to filter by. | Optional | 
| scan_ids | Comma-separated list of scan IDs to filter by. | Optional | 
| status | Comma-separated list of scan statuses to filter by. | Optional | 
| started_on | UTC-format time of scan start to filter by. | Optional | 
| completed_on | UTC-format time of scan completion to filter by. | Optional | 

> | offset | Starting index of overall result set from which to return ids. | Optional | 

Reviewer 2: ```suggestion
| offset | Starting index of overall result set from which to return IDs. | Optional | 
```

| limit | Maximum number of resources to return. | Optional | 

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |

> | CrowdStrike.ODSScanHost.id | String | a unique identifier for the scan event. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.id | String | A unique identifier for the scan event. | 
```

> | CrowdStrike.ODSScanHost.cid | String | a unique identifier for the client that triggered the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.cid | String | A unique identifier for the client that triggered the scan. | 
```

> | CrowdStrike.ODSScanHost.scan_id | String | a unique identifier for the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.scan_id | String | A unique identifier for the scan. | 
```

> | CrowdStrike.ODSScanHost.profile_id | String | a unique identifier for the scan profile used in the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.profile_id | String | A unique identifier for the scan profile used in the scan. | 
```

> | CrowdStrike.ODSScanHost.host_id | String | a unique identifier for the host that was scanned | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.host_id | String | A unique identifier for the host that was scanned | 
```

> | CrowdStrike.ODSScanHost.host_scan_id | String | a unique identifier for the scan that was performed on the host. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.host_scan_id | String | A unique identifier for the scan that was performed on the host. | 
```

> | CrowdStrike.ODSScanHost.filecount.scanned | Number | the number of files that were scanned during the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.filecount.scanned | Number | The number of files that were scanned during the scan. | 
```

> | CrowdStrike.ODSScanHost.filecount.malicious | Number | the number of files that were detected as malicious during the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.filecount.malicious | Number | The number of files that were detected as malicious during the scan. | 
```

> | CrowdStrike.ODSScanHost.filecount.quarantined | Number | the number of files that were quarantined during the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.filecount.quarantined | Number | The number of files that were quarantined during the scan. | 
```

> | CrowdStrike.ODSScanHost.filecount.skipped | Number | the number of files that were skipped during the scan. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.filecount.skipped | Number | The number of files that were skipped during the scan. | 
```

> | CrowdStrike.ODSScanHost.status | String | the status of the scan. \(e.g., "completed", "pending", "cancelled", "running", or "failed"\) | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.status | String | The status of the scan \(e.g., "completed", "pending", "cancelled", "running", or "failed"\) | 
```

> | CrowdStrike.ODSScanHost.severity | Number | a severity score assigned to the scan, ranging from 0 to 100 | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.severity | Number | A severity score assigned to the scan, ranging from 0 to 100 | 
```

> | CrowdStrike.ODSScanHost.started_on | Date | the date and time when the scan was started. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.started_on | Date | The date and time when the scan was started. | 
```

> | CrowdStrike.ODSScanHost.completed_on | Date | the date and time when the scan was completed. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.completed_on | Date | The date and time when the scan was completed. | 
```

> | CrowdStrike.ODSScanHost.last_updated | Date | the date and time when the scan event was last updated. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScanHost.last_updated | Date | The date and time when the scan event was last updated. | 
```

                "last_updated": "2023-05-07T08:28:56.856575358Z",
                "profile_id": "a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1",
                "scan_id": "a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1",
                "started_on": "2023-05-07T08:25:48.336234188Z",
                "status": "completed"
            }
        ]
    }
}
```

#### Human Readable Output

>### CrowdStrike Falcon ODS Scan Hosts

>|ID|Scan ID|Host ID|Filecount|Status|Severity|Started on|
>|---|---|---|---|---|---|---|
>| a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 |  | pending |  |  |
>| a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1 | scanned: 0<br/>malicious: 0<br/>quarantined: 0<br/>skipped: 0<br/>traversed: 524581 | completed |  | 2023-05-07T08:25:48.336234188Z |

### cs-falcon-ods-query-malicious-files

***
Retrieve ODS malicious file details.

#### Base Command

`cs-falcon-ods-query-malicious-files`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| filter | Valid CS-Falcon-FQL filter to query with. | Optional | 

> | file_ids | Comma-separated list of malicious file IDs to retrieve details about. If set will override all other arguments. | Optional | 

Reviewer 2: ```suggestion
| file_ids | Comma-separated list of malicious file IDs to retrieve details about. If set, will override all other arguments. | Optional | 
```

| host_ids | Comma-separated list of host IDs to filter by. | Optional | 
| scan_ids | Comma-separated list of scan IDs to filter by. | Optional | 
| file_paths | Comma-separated list of file paths to filter by. | Optional | 
| file_names | Comma-separated list of file names to filter by. | Optional | 
| hash | Comma-separated list of hashes to filter by. | Optional | 

> | offset | Starting index of overall result set from which to return ids. | Optional | 

Reviewer 2: ```suggestion
| offset | Starting index of overall result set from which to return IDs. | Optional | 
```

| limit | Maximum number of resources to return. | Optional | 

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |

> | CrowdStrike.ODSMaliciousFile.id | String | a unique identifier of the detection event. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.id | String | A unique identifier of the detection event. | 
```

> | CrowdStrike.ODSMaliciousFile.cid | String | a unique identifier for the client that triggered the detection event. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.cid | String | A unique identifier for the client that triggered the detection event. | 
```

> | CrowdStrike.ODSMaliciousFile.scan_id | String | a unique identifier for the scan that triggered the detection event. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.scan_id | String | A unique identifier for the scan that triggered the detection event. | 
```

> | CrowdStrike.ODSMaliciousFile.host_id | String | a unique identifier for the scan that detected the file on the host. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.host_id | String | A unique identifier for the scan that detected the file on the host. | 
```

> | CrowdStrike.ODSMaliciousFile.host_scan_id | String | a unique identifier for the scan that detected the file on the host. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.host_scan_id | String | A unique identifier for the scan that detected the file on the host. | 
```

> | CrowdStrike.ODSMaliciousFile.filepath | String | the full path to the malicious file on the host system. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.filepath | String | The full path to the malicious file on the host system. | 
```

> | CrowdStrike.ODSMaliciousFile.filename | String | the name of the malicious file. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.filename | String | The name of the malicious file. | 
```

> | CrowdStrike.ODSMaliciousFile.hash | String | a SHA-256 hash of the malicious file, which can be used to identify it. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.hash | String | A SHA-256 hash of the malicious file, which can be used to identify it. | 
```

> | CrowdStrike.ODSMaliciousFile.pattern_id | Number | the identifier of the pattern used to detect the malicious file. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.pattern_id | Number | The identifier of the pattern used to detect the malicious file. | 
```

> | CrowdStrike.ODSMaliciousFile.severity | Number | a severity score assigned to the detection event, ranging from 0 to 100. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.severity | Number | A severity score assigned to the detection event, ranging from 0 to 100. | 
```

> | CrowdStrike.ODSMaliciousFile.quarantined | Boolean | a Boolean value indicating whether the file has been quarantined. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.quarantined | Boolean | A Boolean value indicating whether the file has been quarantined. | 
```

> | CrowdStrike.ODSMaliciousFile.last_updated | Date | the date and time when the detection event was last updated. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSMaliciousFile.last_updated | Date | The date and time when the detection event was last updated. | 
```

`cs-falcon-ods-create-scan`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| hosts | List of hosts to be scanned. "hosts" OR "host_groups" must be set. | Optional | 
| host_groups | List of host groups to be scanned. "hosts" OR "host_groups" must be set. | Optional | 
| file_paths | List of file paths to be scanned. "file_paths" OR "scan_inclusions" must be set. | Optional | 
| scan_inclusions | List of included files or locations for this scan. "file_paths" OR "scan_inclusions" must be set. | Optional | 
| scan_exclusions | List of excluded files or locations for this scan. | Optional | 
| initiated_from | Scan origin. | Optional | 
| cpu_priority | Set the scan CPU priority. Possible values are: Highest, High, Medium, Low, Lowest. Default is Low. | Optional | 
| description | Scan description. | Optional | 
| quarantine | Flag indicating if identified threats should be quarantined. | Optional | 
| pause_duration | Amount of time (in seconds) for scan pauses. Default is 2. | Optional | 
| sensor_ml_level_detection | Sensor ML detection level. | Optional | 
| sensor_ml_level_prevention | Sensor ML prevention level. | Optional | 

> | cloud_ml_level_detection | ML Detection level for the scan. | Optional | 

Reviewer 2: ```suggestion
| cloud_ml_level_detection | Cloud ML detection level for the scan. | Optional | 
```

> | cloud_ml_level_prevention | ML Prevention level for the scan. | Optional | 

Reviewer 2: ```suggestion
| cloud_ml_level_prevention | Cloud ML prevention level for the scan. | Optional | 
```

| max_duration | Maximum time (in seconds) the scan is allowed to execute. Default is 2. | Optional | 
| interval_in_seconds | The interval in seconds between each poll. Default is 30. | Optional | 
| timeout_in_seconds | The timeout in seconds until polling ends. Default is 600. | Optional | 

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |
| CrowdStrike.ODSScan.id | String | A unique identifier for the scan event. | 
| CrowdStrike.ODSScan.cid | String | A unique identifier for the client that triggered the scan. | 
| CrowdStrike.ODSScan.profile_id | String | A unique identifier for the scan profile used in the scan. | 
| CrowdStrike.ODSScan.description | String | the ID of the description of the scan. | 
| CrowdStrike.ODSScan.scan_inclusions | String | The files or folders included in the scan. | 
| CrowdStrike.ODSScan.initiated_from | String | The source of the scan initiation. | 

> | CrowdStrike.ODSScan.quarantine | Boolean | If was set to quarantine. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.quarantine | Boolean | Whether the scan was set to quarantine. | 
```

| CrowdStrike.ODSScan.cpu_priority | Number | The CPU priority for the scan \(1-5\). | 
| CrowdStrike.ODSScan.preemption_priority | Number | The preemption priority for the scan. | 
| CrowdStrike.ODSScan.metadata.host_id | String | A unique identifier for the host that was scanned. | 
| CrowdStrike.ODSScan.metadata.host_scan_id | String | A unique identifier for the scan that was performed on the host. | 
| CrowdStrike.ODSScan.metadata.scan_host_metadata_id | String | A unique identifier for the metadata associated with the host scan. | 
| CrowdStrike.ODSScan.metadata.filecount.scanned | Number | The number of files that were scanned. | 
| CrowdStrike.ODSScan.metadata.filecount.malicious | Number | The number of files that were identified as malicious. | 
| CrowdStrike.ODSScan.metadata.filecount.quarantined | Number | The number of files that were quarantined. | 
| CrowdStrike.ODSScan.metadata.filecount.skipped | Number | The number of files that were skipped during the scan. | 
| CrowdStrike.ODSScan.metadata.filecount.traversed | Number | The number of files that were traversed during the scan. | 

> | CrowdStrike.ODSScan.metadata.status | String | The status of the scan on this host. \(e.q., "pending", "running", "completed", or "failed"\). | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.metadata.status | String | The status of the scan on this host \(e.g., "pending", "running", "completed", or "failed"\). | 
```

| CrowdStrike.ODSScan.metadata.started_on | Date | The date and time that the scan started. | 
| CrowdStrike.ODSScan.metadata.completed_on | Date | The date and time that the scan completed. | 
| CrowdStrike.ODSScan.metadata.last_updated | Date | The date and time that the metadata was last updated. | 

> | CrowdStrike.ODSScan.status | String | The status of the scan. \(e.q., "pending", "running", "completed", or "failed"\). | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScan.status | String | The status of the scan \(e.g., "pending", "running", "completed", or "failed"\). | 
```

| CrowdStrike.ODSScan.hosts | String | A list of the host IDs that were scanned. | 
| CrowdStrike.ODSScan.endpoint_notification | Boolean | A boolean value indicating whether endpoint notifications are enabled. | 

> | CrowdStrike.ODSScan.pause_duration | Number | The number of minutes to pause between scanning each file in hours. | 

Reviewer 2: Not sure about this description.
Its the "number of minutes", but what does it mean "in hours"?

### cs-falcon-ods-create-scheduled-scan

***
Create an ODS scheduled scan.

#### Base Command

`cs-falcon-ods-create-scheduled-scan`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| host_groups | List of host groups to be scanned. | Required | 
| file_paths | List of file paths to be scanned. "file_paths" OR "scan_inclusions" must be set. | Optional | 
| scan_inclusions | List of included files or locations for this scan. "file_paths" OR "scan_inclusions" must be set. | Optional | 
| scan_exclusions | List of excluded files or locations for this scan. | Optional | 
| initiated_from | Scan origin. | Optional | 
| cpu_priority | Set the scan CPU priority. Possible values are: Highest, High, Medium, Low, Lowest. Default is Low. | Optional | 
| description | Scan description. | Optional | 
| quarantine | Flag indicating if identified threats should be quarantined. | Optional | 
| pause_duration | Amount of time (in seconds) for scan pauses. Default is 2. | Optional | 
| sensor_ml_level_detection | Sensor ML detection level. | Optional | 
| sensor_ml_level_prevention | Sensor ML prevention level. | Optional | 

> | cloud_ml_level_detection | ML Detection level for the scan. | Optional | 

Reviewer 2: ```suggestion
| cloud_ml_level_detection | Cloud ML detection level for the scan. | Optional | 
```

> | cloud_ml_level_prevention | ML Prevention level for the scan. | Optional | 

Reviewer 2: ```suggestion
| cloud_ml_level_prevention | Cloud ML prevention level for the scan. | Optional | 
```

| CrowdStrike.ODSScheduledScan.pause_duration | Number | The pause duration of scan in hours. | 
| CrowdStrike.ODSScheduledScan.max_duration | Number | The max duration of scan in hours. | 
| CrowdStrike.ODSScheduledScan.max_file_size | Number | The maximum file size that the scan can handle in MB. | 
| CrowdStrike.ODSScheduledScan.sensor_ml_level_detection | Number | The machine learning detection level for the sensor. | 
| CrowdStrike.ODSScheduledScan.cloud_ml_level_detection | Number | The machine learning detection level for the cloud. | 
| CrowdStrike.ODSScheduledScan.schedule.start_timestamp | Date | The timestamp when the first scan was created. | 
| CrowdStrike.ODSScheduledScan.schedule.interval | Number | The interval between scans. | 
| CrowdStrike.ODSScheduledScan.created_on | Date | The timestamp when the scan was created. | 
| CrowdStrike.ODSScheduledScan.created_by | String | The user who created the scan. | 
| CrowdStrike.ODSScheduledScan.last_updated | Date | The timestamp when the scan was last updated. | 
| CrowdStrike.ODSScheduledScan.deleted | Boolean | Whether the scan has been deleted. | 

> | CrowdStrike.ODSScheduledScan.quarantine | Boolean | If was set to quarantine. | 

Reviewer 2: ```suggestion
| CrowdStrike.ODSScheduledScan.quarantine | Boolean | Whether the scan was set to quarantine. | 
```
